src/components/ReviewChats/ChatRequestMutators.tsx

```diff
@@ -0,0 +1,122 @@
+import { Fragment, useState } from "react";
+import { useCancelChatRequestMutation } from "../../generated/graphql";
+import { Input, Text } from "../atomic";
+import CatchUnsavedChangesModal from "../CatchUnsavedChangesModal";
+import SelectOptionModal from "../SelectOptionModal";
+
+type ChatMutatorsProps = {
+  chatRequestId: string;
+  chatCanceled: boolean;
+};
+
+const ChatMutators = ({ chatRequestId, chatCanceled }: ChatMutatorsProps) => {
+  const [clicked, setClicked] = useState(false);
+  const [action, setAction] = useState("");
+  const [cancelReason, setCancelReason] = useState("");
+  const [modified, setModified] = useState(false);
+  const [cancelChatRequestMutation] = useCancelChatRequestMutation();
+
+  console.log(chatCanceled);
+  return (
+    <Fragment>
+      {/* This catch is super incomplete bc it doesn't prevent users from closing the modal mid-editting */}
+      <CatchUnsavedChangesModal unsavedChangesExist={modified === true} />
+      {!chatCanceled && (
+        <button
+          onClick={() => {
+            setClicked(true);
+            setAction("cancel");
+          }}
+        >
+          <Text error u>
+            Cancel
+          </Text>
+        </button>
+      )}
+      <SelectOptionModal
+        isOpen={clicked}
+        onClose={() => {
+          setClicked(false);
+          setAction("");
+        }}
+        title={`What is your reason for ${action}ing?`}
+        onPrimaryButtonClick={() => {
+          if (action === "cancel")
+            cancelChatRequestMutation({
+              variables: {
+                chatRequestId: chatRequestId,
+                chatCancelMessage: cancelReason,
+              },
+            }); //Cancel query
+          if (action === "reschedule") console.log("Reschedule not ready yet"); //Reschedule query
+          setClicked(false);
+        }}
+        onSecondaryButtonClick={() => {
+          setClicked(false);
+        }}
+        primaryButtonText="Submit"
+        secondaryButtonText="Close"
+      >
+        <Input
+          placeholder={"Reason for " + action + "ing (optional)"}
```

Comment:
I'm not sure how will this will generalize

Reply:
It is supposed to be reused when rescheduling. In such a case, `action == "rechedule"`

Reply:
if that's kind of weird, it can be thrown out tho